### PR TITLE
feat(coderd): add parameter to force health check in /debug/health

### PR DESCRIPTION
**Pull request by johnstcn**
Part of https://github.com/coder/coder/issues/8969
Modifies the `/debug/health` endpoint to accept a `force=(true|_)` parameter.
Iff set to `true`, will re-run the healthcheck immediately.
UI changes will come in a separate PR.

### CODE REVIEW

File: coderd/debug.go [hunk 1]
```diff
@@ -41,16 +41,22 @@ func (api *API) debugTailnet(rw http.ResponseWriter, r *http.Request) {
...
-	if report := api.healthCheckCache.Load(); report != nil {
-		if time.Since(report.Time) < api.Options.HealthcheckRefresh {
-			formatHealthcheck(ctx, rw, r, report)
-			return
+	// Check if the forced query parameter is set.
+	forced := r.URL.Query().Get("force") == "true"
+
+	// Get cached report if it exists and the requester did not force a refresh.
+	if !forced {
+		if report := api.healthCheckCache.Load(); report != nil {
```
Comment (mtojek) [inline coderd/debug.go:55]: Do we need an extra safety check to ensure that only 1 request is in-flight?

File: coderd/debug.go [hunk 2]
```diff
@@ -41,16 +41,22 @@ func (api *API) debugTailnet(rw http.ResponseWriter, r *http.Request) {
...
 	ctx, cancel := context.WithTimeout(r.Context(), api.Options.HealthcheckTimeout)
 	defer cancel()
 
-	// Get cached report if it exists.
-	if report := api.healthCheckCache.Load(); report != nil {
-		if time.Since(report.Time) < api.Options.HealthcheckRefresh {
-			formatHealthcheck(ctx, rw, r, report)
-			return
+	// Check if the forced query parameter is set.
+	forced := r.URL.Query().Get("force") == "true"
```
Comment (mtojek) [inline coderd/debug.go:51]: I get the invalidation concept, but if it does not take too much time, maybe we can also run the health check without caching?

**Review by mtojek — COMMENTED** [review]
This is one of these cases, when unit tests are longer than actual implementation.

File: coderd/debug.go (hunk 1 above)
Comment (johnstcn) [reply coderd/debug.go:55]: I belive the healthcheck singleflight group should take care of that.

**Review by mtojek — APPROVED** [review]
nothing blocking here 👍

File: coderd/debug.go (hunk 2 above)
Comment (johnstcn) [reply coderd/debug.go:51]: > if it does not take too much time

I think this is a pretty big 'if' :-)

**Pull request merged**